GitCommands/Git/GitRef.cs

```diff
@@ -126,7 +126,7 @@ string ParseName()
 
         public bool IsOther => !IsHead && !IsRemote && !IsTag;
 
-        public string LocalName => IsRemote ? Name.Substring(Remote.Length + 1) : Name;
+        public string LocalName => IsRemote && Name.Length > Remote.Length + 1 ? Name.Substring(Remote.Length + 1) : Name;
```

Comment:
OK - but perhaps ❔
```suggestion
        public string LocalName => IsRemote && Name.StartsWith($"{Remote}/") ? Name.Substring(Remote.Length + 1) : Name;
```

Reply:
I thought about that too. A good enough reason is that the testcases and names were weird.
So I changed this instead of answering any issues tonight.

Unless there are objections, I plan to squash merge tomorrow or so.

Reply:
Renamed the testcases some more, no other changes